### ISSUE TITLE
extrude with sin function as scale behaves strangely

### ISSUE THREAD

**Issue by canadaduane** [colah/ImplicitCAD#58]
I am trying to create an "undulating cylinder" of some kind, perhaps like a fancy chair leg or something.  The thought was to use a 2D circle and extrude it with a varying scale function, a simple sin function to start with. Strange things ensued.

![Pleasant3D Screenshot](https://dl.dropbox.com/u/118766/Ultimaker/implicitcad_extrude_scale1.png)

![Blender Screenshot](https://dl.dropbox.com/u/118766/Ultimaker/implicitcad_extrude_scale2.png)

```
linear_extrude(height=20, center=true, scale(h)=0.5+cos(h)) {
  circle(10);
}
```

**Comment by julialongtin** [comment]
could you provide new screenshots, and re-test?

**Comment by canadaduane** [comment]
Aside from the poor render quality, this appears to work now.

![image](https://cloud.githubusercontent.com/assets/129/23736587/8830adda-044a-11e7-9987-06fc5e1155bd.png)

**Comment by julialongtin** [comment]
have you tried dialing up the render quality by passing a smaller value
with -r on the command line?

**Comment by canadaduane** [comment]
Here's with `-r 0.5`

![image](https://cloud.githubusercontent.com/assets/129/23757394/f398b32a-04a3-11e7-9c36-7b06cf476a64.png)

Here's with `-r 0.2` (10 sec. render time)

![image](https://cloud.githubusercontent.com/assets/129/23757455/2301e26c-04a4-11e7-93ce-e03a8fb804ab.png)

I tried `-r 0.1` but it took longer than a minute, so I gave up. That said, `-r 0.2` is pretty good. It would be nice if we had a more precise algorithm for low-resolution though.
